stl/inc/ranges

```diff
@@ -728,6 +735,441 @@ namespace ranges {
+                _STL_VERIFY(_Current != _RANGES end(_Parent->_Range), "cannot dereference end transform_view iterator");
+            }
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+
+        public:
+            using iterator_concept = conditional_t<random_access_range<_Vw>, random_access_iterator_tag,
+                conditional_t<bidirectional_range<_Vw>, bidirectional_iterator_tag,
+                    conditional_t<forward_range<_Vw>, forward_iterator_tag, input_iterator_tag>>>;
+            using value_type       = remove_cvref_t<invoke_result_t<_Fn&, range_reference_t<_Base>>>;
+            using difference_type  = range_difference_t<_Base>;
+
+            _Iterator() = default;
+
+            constexpr _Iterator(_Parent_t& _Parent_, iterator_t<_Base> _Current_) noexcept(
+                noexcept(is_nothrow_move_constructible_v<iterator_t<_Base>>)) // strengthened
+                : _Current{_STD move(_Current_)}, _Parent{_STD addressof(_Parent_)} {}
+
+            // clang-format off
+            constexpr _Iterator(_Iterator<!_Const> _It)
+                noexcept(is_nothrow_constructible_v<iterator_t<_Base>, iterator_t<_Vw>>) // strengthened
+                requires _Const && convertible_to<iterator_t<_Vw>, iterator_t<_Base>>
+                : _Current{_STD move(_It._Current)}, _Parent{_It._Parent} {}
+            // clang-format on
+
+            _NODISCARD constexpr iterator_t<_Base> base() const& requires copyable<iterator_t<_Base>> {
+                return _Current;
+            }
+            _NODISCARD constexpr iterator_t<_Base> base() && {
+                return _STD move(_Current);
+            }
+
+            _NODISCARD constexpr decltype(auto) operator*() const {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _Check_dereference();
+                _STL_VERIFY(
+                    _Parent->_Fun, "Cannot dereference iterator into transform_view with no transformation function");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                return _STD invoke(*_Parent->_Fun, *_Current);
+            }
+
+            constexpr _Iterator& operator++() {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                _STL_VERIFY(_Parent != nullptr, "Cannot increment value-initialized transform_view iterator");
+                _STL_VERIFY(
+                    _Current != _RANGES end(_Parent->_Range), "Cannot increment transform_view iterator past end");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                ++_Current;
+                return *this;
+            }
+
+            constexpr decltype(auto) operator++(int) {
+                if constexpr (forward_range<_Base>) {
+                    auto _Tmp = *this;
+                    ++*this;
+                    return _Tmp;
+                } else {
+                    ++*this;
+                }
+            }
+
+            constexpr _Iterator& operator--() requires bidirectional_range<_Base> {

```

Comment:
I will not pester you with any more noexcept improvements anymore.

Reply:
Strengthened just about everything, I think.